src/components/extensive/Modal/ModalAdd.tsx

```diff
@@ -67,6 +71,54 @@ const ModalAdd: FC<ModalAddProps> = ({
       setFile(files);
     }
   }, [files, setFile]);
+
+  const handleFileChange = (files: File[]) => {
+    const formatFile = (file: File): UploadedFile => ({
+      title: file.name,
+      file_name: file.name,
+      mime_type: file.type,
+      collection_name: "storybook",
+      size: file.size,
+      url: "https://google.com",
+      created_at: "now",
+      uuid: file.name,
+      is_public: true,
+      rawFile: file
+    });
+
+    const acceptedFileTypes = acceptedTYpes
+      ? acceptedTYpes
+          .join(",")
+          .split(",")
+          .map(type => type.trim())
+      : [];
+
+    const filteredFiles = files.filter(file => {
+      if (
+        acceptedFileTypes.length > 0 &&
+        !acceptedFileTypes.some(type => file.type === type || file.name.endsWith(type))
+      ) {
+        setErrorMessage?.(`Unsupported file type. Please upload SHP (zipped), KML, or GeoJSON files only`);
+        return false;
```

Comment:
It seems like if we somehow end up with 0 accepted file types, that should result in an error. It's also weird to me that the error message has the set of accepted types hardcoded when it's being passed in as a prop on the component.

Reply:
When no accepted type is specified, it can accept any type. and the error message now shows the accepted type dinamically.